### ISSUE TITLE
The "Unavailable" value is not a valid IP address. 

### ISSUE THREAD

**Issue by KallivdH** [phptek/silverstripe-sentry#92]
Maybe related to #91. This error came after I updated to 5.0.3

[Sentry Issue](https://directlease.sentry.io/share/issue/3493097647f147398ad7f0f064499751/)

**Comment by dcentrica** [comment]
Looks like `UserDataBag` (silly name) requires an actual IP address somewhere. The module could just default to `0.0.0.0` which Wikipedia says can be used as in-place of a real IP when an IP is unknown.

**Comment by dcentrica** [comment]
@KallivdH Have a look at the `master`` branch with a fix in it. If it works, it'll get tagged `5.0.4` shortly.

**Comment by phptek** [comment]
@KallivdH Does the fix mentioned above work for you? If so, I'm keen to close this issue and issue a new release.

**Comment by KallivdH** [comment]
Hey, we use sentry only on our production environments. Unfortunately, there isn't an easy way to test this.

**Comment by phptek** [comment]
Closing as "Assumed Fixed"

**Comment by micschk** [comment]
**Still reproducible on 5.0.3 (SS5 / PHP 8.3) — requesting a 5.0.4 backport**

Confirming this is still present in `5.0.3`. Root cause in `src/Log/SentryLogger.php`: under CLI there's no current `Controller` and no `$_SERVER['REMOTE_ADDR']`, so `get_ip()` returns `''`, which `'ip_address' => self::get_ip() ?: self::SLW_NOOP` turns into the string `"Unavailable"`. Sentry's `UserDataBag` then validates it as an IP and throws `InvalidArgumentException: The "Unavailable" value is not a valid IP address.`

Why it's nasty in 5.x:
- It throws on **every** log call made from CLI — so any queued job / `dev/task` / cron dies the moment it logs (a job can die on its own error-reporting call), and the IP exception **masks the original error** (the trace points at Monolog/Sentry, not the real cause).
- It fires **even with `SENTRY_DSN` unset** — `_config/config.yml` pushes `SentryHandler` onto `Psr\Log\LoggerInterface` unconditionally, so "Sentry is dormant here" is no protection.

I can see this is fixed on `master`/`6.0.0` (`get_ip(): ?string` → `null`, `?? self::DEFAULT_IP`) — but `6.0.0` requires `silverstripe/framework: ^6`, and there's no `5.0.4`, so SS5 sites can't get the fix. The change has no SS6 dependency — **could it be backported to a `5.0.4`?**

Workaround for other 5.x users, in project `_config.php`:
```php
if (PHP_SAPI === 'cli' && !isset($_SERVER['REMOTE_ADDR'])) {
    $_SERVER['REMOTE_ADDR'] = '127.0.0.1';
}
```

**Comment by dcentrica** [comment]
@micschk Nice find! I've backported the fix and released `5.0.4`. :-)
